app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -505,8 +597,52 @@ class _CustomChatInputState extends ConsumerState<CustomChatInput> {
         GestureDetector(
           onTap: () {
             inputNotifier.toggleReplyView(false);
+            inputNotifier.toggleEditView(false);
+            inputNotifier.setReplyWidget(null);
+            inputNotifier.setEditWidget(null);
+            inputNotifier.setRepliedToMessage(null);
+            inputNotifier.setEditMessage(null);
+            FocusScope.of(context).unfocus();
+          },
+          child: const Icon(
+            Atlas.xmark_circle,
+            color: Colors.white,
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget editMessageBuilder(
+    BuildContext context,
+    WidgetRef ref,
+    Widget? child,
+  ) {
+    final roomId = widget.convo.getRoomIdStr();
+    final inputNotifier = ref.watch(chatInputProvider(roomId).notifier);
+    return Row(
+      children: [
+        const SizedBox(width: 5),
+        const Text(
+          'Edit:',
```

Comment:
Minor typo, should be `Editing Message`.

Reply:
I think `Editing Message` will not be good typo as this is not a label for some process indication. This is the instruction label and that's why I have used just `Edit`.

I have checked other app also for this. Telegram is using just `Edit Message` label.

Telegram Reference Image:
![Telegram](https://github.com/acterglobal/a3/assets/51526480/45f7f9ef-f9c4-4fc3-a2d9-398e540aa7ad)